lib/datadog/tracing/contrib/roda/instrumentation.rb

```diff
@@ -0,0 +1,75 @@
+require_relative '../../metadata/ext'
+require_relative 'ext'
+require_relative '../analytics'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Roda
+        # Instrumentation for Roda
+        module Instrumentation
+
+          def _roda_handle_main_route
+            instrument(Ext::SPAN_REQUEST) { super }
+          end
+
+          def call
+            instrument(Ext::SPAN_REQUEST) { super }
+          end
+
+          private
+
+          def instrument(span_name, &block)
+            set_distributed_tracing_context!(request.env)
+
+            Tracing.trace(span_name) do |span|
+              begin
+                request_method = request.request_method.to_s.upcase
+
+                span.service = configuration[:service_name]
+                span.span_type = Tracing::Metadata::Ext::HTTP::TYPE_INBOUND
+
+                # Using the http method as a resource, since the URL/path can trigger
+                # a possibly infinite number of resources.
+                span.resource = request_method
+
+                span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_URL, request.path)
+                span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_METHOD, request_method)
+
+                # Add analytics tag to the span
+                if Contrib::Analytics.enabled?(configuration[:analytics_enabled])
+                  Contrib::Analytics.set_sample_rate(span, configuration[:analytics_sample_rate])
+                end
+
+              # Measure service stats
+              Contrib::Analytics.set_measured(span)
+
+              ensure
+                response = yield
+              end
+              
+              status_code = response[0]
+
+              # Adds status code to the resource name once the resource comes back
+              span.resource = "#{request_method} #{status_code}"
```

Comment:
For unexpected errors (e.g. `raise 'oops'`), this part of the code here is likely unreachable.
We should try to match the response code that Roda returns to the HTTP client in that scenario.